File: solutions/lua/bob/1/bob.lua
local bob = {}

function is_silence(phrase)
  return string.match(phrase, "^%s*$")
end

function is_question(phrase)
  return string.match(phrase, "^.*%?%s*$")
end

function is_shouting(phrase)
  return string.match(phrase, "%u") and not string.match(phrase, "%l")
end

function bob.hey(say)
  if is_silence(say) then return "Fine, be that way." end

  if is_question(say) and is_shouting(say) then
    return "Calm down, I know what I'm doing!"
  end

  if is_question(say) then return "Sure" end
  if is_shouting(say) then return "Whoa, chill out!" end

  return "Whatever"
end

return bob

File: solutions/javascript/bob/1/bob.js
const isSilence = (message) => !message.match(/[^\s]+/g);

const isShouting = (message) => {
  return (
        message.match(/[A-Z]+/g)
    && !message.match(/[a-z]+/g)
  );
};

const isQuestion = (message) => message.trim().endsWith('?');

export const hey = (message) => {
  if (isSilence(message)) {
    return 'Fine. Be that way!';
  }

  if (isShouting(message) && isQuestion(message)) {
    return "Calm down, I know what I'm doing!"
  }

  if (isShouting(message)) {
    return 'Whoa, chill out!';
  }

  if (isQuestion(message)) {
    return 'Sure.'
  }

  return 'Whatever.';
};
